cantabular/client.go

```diff
@@ -13,23 +13,40 @@ import (
 	"github.com/ONSdigital/dp-api-clients-go/v2/health"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/log.go/v2/log"
+
+	"github.com/shurcooL/graphql"
 )
 
 // Service is the cantabular service name
 const Service = "cantabular"
 
 // Client is the client for interacting with the Cantabular API
 type Client struct {
-	ua   httpClient
-	host string
+	ua         httpClient
+	gqlClient  GraphQLClient
+	host       string
+	extApiHost string
 }
 
 // NewClient returns a new Client
-func NewClient(ua httpClient, cfg Config) *Client {
-	return &Client{
-		ua:   ua,
-		host: cfg.Host,
+func NewClient(cfg Config, ua httpClient, g GraphQLClient) *Client {
```

Comment:
Just a suggestion, maybe we can create extra constructors?
e.g. 
- `NewClient(cfg Config) *Client` : creates an httpClient and a GraphQLClient
- `NewClientWithHttp(cfg Config, ua httpClient) *Client` : uses the provided httpClient and creates a new GraphQLClient
- `NewClientWithHttpAndGrapQL(cfg Config, ua httpClient, g GraphQLClient) *Client` : uses the provided httpClient and GraphQLClient

Then we could also create getters for the httpClient and the GraphQLCLient, so that they can be reused by the caller if required

Do you think it would be a good approach?

Reply:
Yeah I went back and forth with how best to do this. Perhaps it couldn't hurt to include extra constructors. I did want to include getters. It will be breaking either way so I'm to go ahead as is and maybe add more a little down the line